sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -105,3 +105,15 @@ case class FunctionIdentifier(funcName: String, database: Option[String])
 object FunctionIdentifier {
   def apply(funcName: String): FunctionIdentifier = new FunctionIdentifier(funcName)
 }
+
+/**
+ * Catalog identifier.
+
+ * @param space Namespace in the catalog
+ * @param name Object name
+ */
+case class CatalogIdentifier(space: CatalogIdentifier.Namespace, name: String)
```

Comment:
I think that these should be in the `org.apache.spark.sql.catalyst.catalog.v2` namespace. That separates the new identifiers from the old ones and keeps them with the other v2 additions.

Also, we will need to translate this to Java. I wrote the SPIP using scala conventions for clarity, but we should use types that are Java-friendly. For example, Java implementations should not be passed `Seq` so namespace should be `Array[String]` instead.